builder.rb

```diff
@@ -87,7 +87,98 @@ def run_nunit_tests(options = nil)
     end
   end
 
+  def run_nunit_lite_tests
+    dir = Dir.mktmpdir
+    begin
+      touch_unit_server = get_touch_unit_server(dir)
+
+      logfile = 'tests.log'
+      test_commands, errors = @analyzer.nunit_light_test_commands(@configuration, @platform, touch_unit_server, logfile)
+
+      if test_commands.nil? || test_commands.empty?
+        errors = ['Failed to create test command'] if errors.empty?
+        raise errors.join("\n")
+      end
+
+      app_file = nil
+      test_commands.each do |test_command|
+        puts
+        puts "\e[34m#{test_command}\e[0m"
+        puts
+
+        command = test_command.join(' ')
+        command.sub! '--launchsim', "--launchsim #{app_file}" if command.include? touch_unit_server and !app_file.nil?
+
+        raise 'Test failed' unless system(command)
+
+        if command.include? 'mdtool'
+          @generated_files = @analyzer.collect_generated_files(@configuration, @platform, @project_type_filter)
+          @generated_files.each do |_, project_output|
+            app_file = project_output[:app] if project_output[:api] == Api::IOS and project_output[:app]
+          end
+        end
+      end
+
+      results = process_touch_unit_logs(logfile)
+
+      # remove logs file
+      FileUtils.remove_entry logfile
+
+      raise 'Test failed' if results['Failed'] != '0'
+    rescue => ex
+      error_with_message(ex.inspect.to_s)
+      error_with_message('--- Stack trace: ---')
+      error_with_message(ex.backtrace.to_s)
+      exit(1)
+    ensure
+      FileUtils.remove_entry dir
+    end
+  end
+
   def generated_files
     @generated_files
   end
+
+  private
+
+  def get_touch_unit_server(directory)
```

Comment:
should not we, preinstall the generated `Touch.Server.exe` on xamarin stacks? instead of git clone and build the `Touch.Server` project? it would decrease the step's runtime.